modules/programs/powerline-go.nix

```
@@ -106,17 +106,38 @@ in {
     };
   };
 
-  config = mkIf (cfg.enable && config.programs.bash.enable) {
-    programs.bash.initExtra = ''
-      function _update_ps1() {
-        local old_exit_status=$?
-        PS1="$(${pkgs.powerline-go}/bin/powerline-go -error $old_exit_status ${commandLineArguments})"
+  config = {
+    programs.bash.initExtra =
+      mkIf (cfg.enable && config.programs.bash.enable) ''
+        function _update_ps1() {
+          local old_exit_status=$?
+          PS1="$(${pkgs.powerline-go}/bin/powerline-go -error $old_exit_status ${commandLineArguments})"
+          ${cfg.extraUpdatePS1}
+          return $old_exit_status
+        }
+
+        if [ "$TERM" != "linux" ]; then
+          PROMPT_COMMAND="_update_ps1;$PROMPT_COMMAND"
```

Can the prompt command end with a trailing semicolon?

If not:

```suggestion
          PROMPT_COMMAND="_update_ps1\${PROMPT_COMMAND:+';'}$PROMPT_COMMAND"
```

---

That code is straight from the [powerline-go README](https://github.com/justjanne/powerline-go).

Tested it to be sure, and the trailing semicolon is no problem.